Minio/V4Authenticator.cs

```diff
@@ -452,8 +452,15 @@ private void SetHostHeader(IRestRequest request, IRestClient client)
         /// Set 'x-amz-content-sha256' http header.
         /// </summary>
         /// <param name="request">Instantiated request object</param>
-        private void SetContentSha256(IRestRequest request)
+        private void SetContentSha256(IRestRequest request, IRestClient client)
         {
+            // No need to compute SHA256 if endpoint scheme is https
+            if (client.BaseUrl.Host.StartsWith("https"))
```

Comment:
We should have the client struct remember if the RestClient is secure or not. It is odd style to passing the entire IRestClient into 'SetContentSha256' and unsafe - as a thumb rule we should only passing the necessary arguments.